### PR TITLE
Formalize the overflow and cardinality limit interaction

### PR DESCRIPTION
**Pull request by reyang**
Follow up based on the TC discussion and spec issue triage discussion: https://github.com/open-telemetry/opentelemetry-specification/issues/3904#issuecomment-1969453098.

Related to #3904 and #3905.

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]
LGTM.

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -759,9 +759,9 @@ independently aggregated because of the limit.
 
 The SDK MUST create an Aggregator with the overflow attribute set
 prior to reaching the cardinality limit and use it to aggregate events
-for which the correct Aggregator could not be created.  The maximum
-number of distinct, non-overflow attributes is one less than the
-limit, as a result.
+for which the correct Aggregator could not be created.  The SDK MUST provide the
+guarantee that overflow does not happen if the the maximum number of distinct,
```
Comment (trask) [inline specification/metrics/sdk.md:763]:
```suggestion
guarantee that overflow does not happen if the maximum number of distinct,
```

File: specification/metrics/sdk.md [hunk 2]
```diff
@@ -759,9 +759,9 @@ independently aggregated because of the limit.
...
+non-overflow attributes is less than or equal to the limit.
```
Comment (trask) [inline specification/metrics/sdk.md:764]:
```suggestion
non-overflow attribute sets is less than or equal to the limit.
```

File: specification/metrics/sdk.md [hunk 3]
```diff
@@ -759,9 +759,9 @@ independently aggregated because of the limit.
 
 The SDK MUST create an Aggregator with the overflow attribute set
 prior to reaching the cardinality limit and use it to aggregate events
-for which the correct Aggregator could not be created.  The maximum
-number of distinct, non-overflow attributes is one less than the
-limit, as a result.
+for which the correct Aggregator could not be created.  The SDK MUST provide the
+guarantee that overflow does not happen if the maximum number of distinct,
```
Comment (utpilla) [inline specification/metrics/sdk.md:763]: nit:

```suggestion
guarantee that overflow would not happen if the maximum number of distinct,
```

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Review by trask — APPROVED** [review]
thanks for the explanation in https://github.com/open-telemetry/opentelemetry-specification/issues/3904#issuecomment-1970349802!

**Pull request merged**